instruments/ondax/lm.py

```diff
@@ -0,0 +1,382 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Provides the support for the Ondax LM Laser.
+
+Class originally contributed by Catherine Holloway.
+"""
+
+# IMPORTS #####################################################################
+
+from __future__ import absolute_import
+from __future__ import division
+
+from enum import IntEnum
+
+import quantities as pq
+
+from instruments.abstract_instruments import Instrument
+from instruments.util_fns import convert_temperature
+
+# CLASSES #####################################################################
+
+
+class LM(Instrument):
+    """
+    The LM is the Ondax SureLock VHG-stabilized laser diode.
+
+    The user manual can be found here:
+    http://www.ondax.com/Downloads/SureLock/Compact%20laser%20module%20manual.pdf
+    """
+
+    def __init__(self, filelike):
+        super(LM, self).__init__(filelike)
+        self.terminator = "\r"
+        self.apc = self.AutomaticPowerControl(self)
+        self.acc = self.AutomaticCurrentControl(self)
+        self.tec = self.ThermoElectricCooler(self)
+        self.modulation = self.Modulation(self)
+
+    # ENUMS #
+    class Status(IntEnum):
+        """
+        Enum containing the states of the laser
+        """
+        normal = 1
+        inner_modulation = 2
+        power_scan = 3
+        calibrate = 4
+        shutdown_current = 5
+        shutdown_overheat = 6
+        waiting_stable_temperature = 7
+        waiting = 8
+
+    # INNER CLASSES #
+
+    class AutomaticCurrentControl(object):
+        """
+        Options and functions relatde to the laser diode's automatic current
+        control driver
+        """
+        def __init__(self, parent):
+            self.parent = parent
+
+        @property
+        def target(self):
+            """
+            The ACC target current
+            :return: Current ACC of the Laser
+            :rtype: quantities.mA
+            """
+            response = float(self.parent.query("rstli?"))
+            return response*pq.mA
+
+        def enable(self):
```

Comment:
I'm thinking about `enable()` and `disable()` both being turned into one `enabled` boolean property. It might make it more consistent with other instruments.